sycl/include/CL/__spirv/spirv_ops.hpp

```diff
@@ -136,100 +160,374 @@ __SPIRV_ATOMICS(__SPIRV_ATOMIC_UNSIGNED, unsigned long long)
-                                            dataT Data) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_SubgroupBlockWriteINTEL(__attribute__((opencl_global)) uint32_t *Ptr,
+                                dataT Data) noexcept;
 
 template <typename dataT>
-extern int32_t __spirv_ReadPipe(RPipeTy<dataT> Pipe, dataT *Data,
-                                int32_t Size, int32_t Alignment) noexcept;
+extern SYCL_EXTERNAL int32_t
+__spirv_ReadPipe(RPipeTy<dataT> Pipe, dataT *Data, int32_t Size,
+                 int32_t Alignment) noexcept;
 template <typename dataT>
-extern int32_t __spirv_WritePipe(WPipeTy<dataT> Pipe, const dataT *Data,
-                                 int32_t Size, int32_t Alignment) noexcept;
+extern SYCL_EXTERNAL int32_t
+__spirv_WritePipe(WPipeTy<dataT> Pipe, const dataT *Data, int32_t Size,
+                  int32_t Alignment) noexcept;
 template <typename dataT>
-extern void __spirv_ReadPipeBlockingINTEL(RPipeTy<dataT> Pipe, dataT *Data,
-                                          int32_t Size,
-                                          int32_t Alignment) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_ReadPipeBlockingINTEL(RPipeTy<dataT> Pipe, dataT *Data, int32_t Size,
+                              int32_t Alignment) noexcept;
 template <typename dataT>
-extern void __spirv_WritePipeBlockingINTEL(WPipeTy<dataT> Pipe,
-                                           const dataT *Data, int32_t Size,
-                                           int32_t Alignment) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_WritePipeBlockingINTEL(WPipeTy<dataT> Pipe, const dataT *Data,
+                               int32_t Size, int32_t Alignment) noexcept;
 template <typename dataT>
-extern RPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_read(
+extern SYCL_EXTERNAL RPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_read(
     const ConstantPipeStorage *Storage) noexcept;
 template <typename dataT>
-extern WPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_write(
+extern SYCL_EXTERNAL WPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_write(
     const ConstantPipeStorage *Storage) noexcept;
 
-extern void __spirv_ocl_prefetch(const __attribute__((opencl_global)) char *Ptr,
-                                 size_t NumBytes) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_ocl_prefetch(const __attribute__((opencl_global)) char *Ptr,
+                     size_t NumBytes) noexcept;
+
+// __attribute((format(...))) enables compiler checks for a format string.
+SYCL_EXTERNAL
+int __spirv_ocl_printf(const __attribute__((opencl_constant)) char *fmt, ...)
+    __attribute__((format(printf, 1, 2)));
+
+#define COMPARISON(Order, Cmp)                                                 \
+  template <typename ReTTT, typename T1, typename T2>                          \
+  extern SYCL_EXTERNAL                                                         \
+  ReTTT __spirv_F##Order##Cmp(T1, T2);
```

Comment:
Just curious, why `ReTTT`? What about `RetTy`?

Reply:
ReTTT was already used somewhere above in this file.

Reply:
Does `ReTTT` look more readable than `RetTy` to you?

Reply:
How about `R` or `RetT`? Why use `Ty` or `TTT` for template arguments?

Reply:
Changed to RetT.